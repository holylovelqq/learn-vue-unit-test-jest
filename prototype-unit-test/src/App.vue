<template>
  <div id="app">
    <el-container>
      <el-header>
        <app-header/>
      </el-header>
      <el-main>
        <router-view/>
      </el-main>
    </el-container>
  </div>
</template>

<script>
import AppHeader from '@/components/common/AppHeader.vue'
export default {
  name: 'App',
  components: {
    AppHeader
  }
}
</script>

<style>
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #2c3e50;
}
.el-header {
  padding: 0;
}
@import 'normalize.css'
</style>
