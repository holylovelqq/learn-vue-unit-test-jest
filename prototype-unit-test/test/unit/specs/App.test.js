import {shallowMount, createLocalVue} from '@vue/test-utils'
import App from '@/App.vue'
import Element from 'element-ui'
import VueRouter from 'vue-router'

const localVue = createLocalVue()
localVue.use(Element)
localVue.use(VueRouter)

describe('App.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(App, {localVue})
  })

  afterEach(() => {
    wrapper.destroy()
  })

  it('App Dom test', () => {
    expect(wrapper.find('#app').exists()).toBe(true)
  })
  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
