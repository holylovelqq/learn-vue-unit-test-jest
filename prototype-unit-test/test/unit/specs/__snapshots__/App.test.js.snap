// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue matches snapshot 1`] = `
<div id="app">
    <elcontainer-stub>
        <elheader-stub height="60px">
            <appheader-stub></appheader-stub>
        </elheader-stub>
        <elmain-stub>
            <router-view-stub name="default"></router-view-stub>
        </elmain-stub>
    </elcontainer-stub>
</div>
`;
